copier/config/user_data.py

```diff
@@ -108,8 +112,36 @@ def cast_answer_type(answer: Any, type_fn: Callable) -> Any:
         return answer
 
 
+def render_string(template: Any, env: SandboxedEnvironment, context: AnyByStrDict):
+    """Render a templated-string using Jinja."""
+    try:
+        return env.from_string(template).render(**context)
+    except TypeError:
+        # template was not a string
+        return template
+    except UndefinedError as error:
+        raise UserMessageError(str(error)) from error
+
+
+def render_choices(choices: Choices, env: SandboxedEnvironment, context: AnyByStrDict):
+    """Render a list of templated choices using Jinja."""
+    render = partial(render_string, env=env, context=context)
+    if isinstance(choices, dict):
+        choices = {render(k): render(v) for k, v in choices.items()}
+    elif isinstance(choices, list):
+        for i, choice in enumerate(choices):
+            if isinstance(choice, (tuple, list)) and len(choice) == 2:
+                choices[i] = (render(choice[0]), render(choice[1]))
+            else:
+                choices[i] = render(choice)
+    return choices
```

Comment:
By using `NativeEnvironment`, all this is very simplified:

```suggestion
    return render(repr(choices))
```

Check this interactive session so you understand what's happening here:

```py
>>> from jinja2.nativetypes import NativeEnvironment                                                                                                                                                              

>>> choices = {"{{ name1 }}": 'value1', '{{ name2 }}':2}                                                                                                                                                          

>>> NativeEnvironment().from_string(repr(choices)).render(name1=0, name2="hey")                                                                                                                                   
{'0': 'value1', 'hey': 2}


>>> choices = ""                                                                                                                                                                                                  

>>> NativeEnvironment().from_string(repr(choices)).render(name1=0, name2="hey")                                                                                                                                   
''


>>> choices = "good"                                                                                                                                                                                              

>>> NativeEnvironment().from_string(repr(choices)).render(name1=0, name2="hey")                                                                                                                                   
'good'
```

Actually it's so much simplified that you could even remove this method.

Reply:
Wow, this is great! Thanks, I'll use that.

Reply:
In fact using `repr` will not work:

```python
>>> from jinja2.nativetypes import NativeEnvironment

>>> nenv = NativeEnvironment(variable_start_string='[[', variable_end_string=']]')

>>> choices = [['one', 'first'], ['two', 'second'], ['three', 'third']]

>>> nenv.from_string(repr(choices)).render()
TemplateSyntaxError: unexpected ']'
```

Reply:
hummmm.

Well, I guess this can't be improved then. Thanks!